core/src/main/java/io/questdb/cairo/O3CopyJob.java

```diff
@@ -496,6 +496,12 @@ private static void copyTail(
                 );
             }
 
+            final int commitMode = tableWriter.getConfiguration().getCommitMode();
+            O3Utils.fsync(ff, srcDataFixFd, commitMode);
```

Comment:
nit: it's worth adding a comment that would clarify why we sync not only the destination but also the source files.

Reply:
this is a good spot, we don't need to sync read-only things, fixed here https://github.com/questdb/questdb/pull/3304/commits/8b45780d47d24e64fdaa790540513f6360a6e30b along with replacing fsync with msync